src/rest/postResetSystem.go

```diff
@@ -0,0 +1,71 @@
+package rest
+
+import (
+	"devicemanager/config"
+	"devicemanager/rest/redfish"
+	"fmt"
+	"github.com/kataras/iris/v12"
+	"github.com/kataras/iris/v12/context"
+	"github.com/sirupsen/logrus"
+	"io"
+	"net/http"
+)
+
+type resetComputerSystemHandler struct {
+	cfg config.Config
+}
+
+func (r *resetComputerSystemHandler) handle(ctx iris.Context) {
+	if ctx.GetHeader("Authorization") == "" {
+		noValidAuthError := "No valid authorization"
+		ctx.StatusCode(http.StatusUnauthorized)
+		logrus.Error(noValidAuthError)
+		ctx.WriteString(noValidAuthError)
+		return
+	}
+
+	var reqInfo redfish.RequestInformation
+
+	// Retrieve request information from a request.
+	err := ctx.ReadJSON(&reqInfo)
+	if err != nil {
+		missingInfoError := "Unable to retrieve mandatory information from a request: " + err.Error()
+		logrus.Error(missingInfoError)
+		ctx.StatusCode(http.StatusBadRequest)
+		ctx.WriteString(missingInfoError)
+		return
+	}
+
+	httpClient := redfish.NewHttpClient(r.cfg).WithBasicAuth(reqInfo.Username, string(reqInfo.Password))
+	systemsUri := fmt.Sprintf("https://%s%s", reqInfo.Host, ctx.Request().RequestURI)
+
+	response, err := httpClient.Post(systemsUri, reqInfo.Body)
```

Comment:
Maybe we can validate the ResetType sending in POST body?

Reply:
Validation added